tests/test_dataset.py

```diff
@@ -643,3 +643,53 @@ def test_to_gemmi_withNans(data_merged):
     roundtrip = rs.DataSet.from_gemmi(data_merged.to_gemmi())
 
     assert pd.isna(roundtrip.loc[data_merged.index[0], "N(+)"])
+
+
+@pytest.mark.parametrize(
+    "dtype",
+    rs.summarize_mtz_dtypes(False)[["MTZ Code", "Name", "Class"]].melt().itertuples(),
+)
+def test_select_mtzdtype(data_hewl, dtype):
```

Comment:
This never tests the situation wherein an `HKL` dtype is outside the index. I am not sure if this matters. I can confirm that it works.

Reply:
I added a `"reset_index", [True, False]` parametrized argument to include HKL outside the index. I agree this makes the test a bit more thorough.